android/compose/src/main/kotlin/com/boswelja/ephemeris/compose/InfiniteHorizontalPager.kt

```diff
@@ -13,57 +14,59 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.layout.layout
 import androidx.compose.ui.layout.onGloballyPositioned
 import androidx.compose.ui.layout.positionInParent
-import androidx.compose.ui.unit.LayoutDirection
 import dev.chrisbanes.snapper.ExperimentalSnapperApi
 import dev.chrisbanes.snapper.rememberLazyListSnapperLayoutInfo
 import dev.chrisbanes.snapper.rememberSnapperFlingBehavior
+import kotlinx.coroutines.flow.filterNot
 
 @OptIn(ExperimentalSnapperApi::class)
 @Composable
-public fun InfiniteHorizontalPager(
+internal fun InfiniteHorizontalPager(
     modifier: Modifier = Modifier,
     state: InfinitePagerState = rememberInfinitePagerState(),
     contentPadding: PaddingValues = PaddingValues(),
     maxItemFling: Int = 1,
     content: @Composable (page: Int) -> Unit
 ) {
-    val internalLazyState = rememberLazyListState(initialFirstVisibleItemIndex = state.internalStartPage)
-    val layoutInfo = rememberLazyListSnapperLayoutInfo(
-        lazyListState = internalLazyState,
-        endContentPadding = contentPadding.calculateEndPadding(LayoutDirection.Ltr)
-    )
+    val layoutInfo = rememberLazyListSnapperLayoutInfo(lazyListState = state.lazyListState)
 
     // Update the pager state page
-    LaunchedEffect(internalLazyState.isScrollInProgress) {
-        if (!internalLazyState.isScrollInProgress) {
-            state.page = layoutInfo.currentItem?.index ?: state.internalStartPage
-        }
+    LaunchedEffect(state.lazyListState) {
+        snapshotFlow { state.lazyListState.isScrollInProgress }
+            .filterNot { it }
+            .collect {
+                state.page = state.calculatePageFromInternal(
+                    layoutInfo.currentItem?.index ?: state.internalPageOffset
+                )
+            }
     }
 
     var pagerHeight by remember { mutableStateOf(0) }
 
     // Switch the Modifier used to animate height once pagerHeight has had it's initial update
     val layoutModifier = if (pagerHeight > 0) {
-        Modifier.animateContentSize().then(modifier)
+        Modifier
+            .animateContentSize()
+            .then(modifier)
     } else modifier
 
     LazyRow(
         modifier = layoutModifier,
-        state = internalLazyState,
+        state = state.lazyListState,
```

Comment:
reading this made me wonder if it was setting its own state because the variable names clash between `LazyRow#state` and `state: InfinitePagerState = rememberInfinitePagerState()` 😩